examples/win32/Win32.cpp

```diff
@@ -3,7 +3,12 @@
 // Headers
 ////////////////////////////////////////////////////////////
 #include <SFML/Graphics.hpp>
+
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
```

Comment:
Is there any advantage of adding this block to all these files instead of just adding it as a compile definition in the CMake configuration? As long as it is declared `PRIVATE` in CMake it should only have an effect when building the library and should not have any effects on user code.

Reply:
There is one place where we cannot use this macro, `SleepImpl.hpp`. I think also the Network module might have a place like that as well. My personal preference would be creating something like `SFML/Priv/WindowsLeanAndMean.hpp` and stick all the repetition there, then just include that header instead. But we don't have a precedent for that. Let me know what you think.

Reply:
SleepImpl.hpp is being removed in https://github.com/SFML/SFML/pull/1872. o in case there's nothing additional in the Network module, having it in CMake should be OK..